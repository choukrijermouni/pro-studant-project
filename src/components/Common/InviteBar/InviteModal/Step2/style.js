import {
  neutral,
  blue
} from '@pro_boa/ui'

export default {
  modal: {
    backgroundColor: neutral[0],
    borderRadius: 4,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'left',
    paddingLeft: 20,
    paddingRight: 20
  },
  licenseButtons: {
    display: 'flex',
    justifyContent: 'space-between',
    marginBottom: 50
  },
  line: {
    backgroundColor: neutral[13],
    height: 1,
    border: 'none',
    marginBottom: 20,
    marginTop: 40,
    width: '100%'
  },
  emailRow: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 20,
    marginTop: 20
  },
  alignAvatar: {
    display: 'flex',
    alignItems: 'center'
  },
  emailsBox: {
    height: 380,
    paddingRight: 12,
    overflow: 'scroll',
    '&::-webkit-scrollbar': {
      width: 5
    },
    '&::-webkit-scrollbar-thumb': {
      background: blue[1],
      borderRadius: 50
    },
    overflowX: 'hidden'
  }
}
