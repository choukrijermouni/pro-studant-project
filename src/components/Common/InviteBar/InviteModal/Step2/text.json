{
    "header": "Veuillez confirmer les coordonn√©es des apprenants",
    "confirmButton": "Confirmer",
    "abortButton": "Annuler"
}