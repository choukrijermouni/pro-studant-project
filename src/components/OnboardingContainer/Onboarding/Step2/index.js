import {
  Paper,
  Button,
  TypographyElement
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import styles from './style'
import { logo as pro_boa } from 'assets/index'
import Text from './text.json'
import { onboardingSteps, OnboardingVideos } from 'constants/'

const useStyle = createUseStyles(styles)

export default ({ setStep }) => {
  const { paper, content, logo, line, licenseButtons, body, video } = useStyle()
  return (
    <div className={body}>
      <Paper className={paper}>
        <div className={content}>
          <img src={pro_boa} alt='notification' className={logo} />
          <TypographyElement
            component='h2'
            variant='heading2'
            align='left'
            spacing='40px 0 0 0'
          >
            {Text.title}
          </TypographyElement>
          <TypographyElement
            component='h4'
            variant='body2'
            align='left'
            spacing='8px 0 26px 0'
          >
            {Text.subHeader.first}{Text.subHeader.second}
          </TypographyElement>
          <video src={OnboardingVideos.OldB2BCustomers} className={video} controls />
          <hr className={line} />
          <div className={licenseButtons}>
            <Button
              handleClick={() => setStep(onboardingSteps.changePassword)}
              width={153}
              label={Text.continueButton}
            />
          </div>
        </div>
      </Paper>
    </div>
  )
}
