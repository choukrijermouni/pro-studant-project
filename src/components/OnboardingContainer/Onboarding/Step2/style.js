import { neutral } from '@pro_boa/ui'
import { headerBackground } from 'assets/index'

export default {
  content: {
    display: 'block',
    padding: '65px 48px'
  },
  logo: {
    height: 38
  },
  line: {
    backgroundColor: neutral[13],
    height: 1,
    border: 'none',
    marginBottom: 20
  },
  licenseButtons: {
    display: 'flex',
    justifyContent: 'end'
  },
  video: {
    width: 800,
    height: 450,
    border: '1px solid #EDEFF9',
    borderRadius: 4,
    boxShadow: '0 0 36px 0 rgba(194,212,230,0.23)',
    marginBottom: 26
  },
  body: {
    backgroundImage: `url(${headerBackground})`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'right top, right top',
    backgroundSize: 270,
    backgroundColor: neutral[0],
    borderRadius: 4,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  paper: {
    display: 'flex',
    flexDirection: 'column'
  }
}
