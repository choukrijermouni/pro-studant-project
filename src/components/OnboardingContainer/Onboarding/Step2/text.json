{
  "title": " Nouvelles fonctionnalités disponibles sur votre nouvel espace Pro",
  "subHeader": {
    "first": "Bonjour",
    "second": ", pro_boa innove à nouveau et lance une nouvelle version de l’espace Pro pour améliorer votre quotidien."
  },
  "continueButton": "Continuer"
}