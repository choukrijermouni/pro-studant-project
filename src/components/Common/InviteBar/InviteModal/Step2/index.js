import {
  TypographyElement,
  SearchInput,
  Avatar,
  AvatarSize,
  Button,
  ButtonVariation,
  ButtonSize
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text.json'
import { avatarName } from 'helpers'
import { useState } from 'react'

const useStyles = createUseStyles(style)
export default ({ learners, cancelEdit, editEmails, closeModal }) => {
  const {
    modal,
    licenseButtons,
    line,
    emailsBox,
    emailRow,
    alignAvatar
  } = useStyles()
  const [email, setEmail] = useState(learners)
  const [searchValue, setSearchValue] = useState('')
  const [filterEmail, setFilterEmail] = useState([])
  const deleteLearner = (targetEmail) => {
    const filteredEmails = email.filter(element => element !== targetEmail)
    setEmail(filteredEmails)
    setSearchValue('')
    setFilterEmail([])
  }
  return (
    <div className={modal}>
      <TypographyElement
        component='h3'
        variant='heading3'
        align='left'
        spacing='60px 0 20px 0'
      >
        {Text.header}
      </TypographyElement>
      <SearchInput
        placeHolder=''
        id='unique_ID_1'
        height={54}
        value={searchValue}
        handleChange={
          (value) => {
            setSearchValue(value)
            setFilterEmail(email.filter(element => element.toLowerCase().search(value.toLowerCase()) >= 0))
          }
        }
      />
      <div className={emailsBox}>
        {
          (searchValue.length || filterEmail.length ? filterEmail : email).map((element, key) =>
            <div key={key} className={emailRow}>
              <div className={alignAvatar}>
                <Avatar
                  img={null}
                  name={avatarName(element)}
                  size={AvatarSize.size_1}
                />
                <TypographyElement
                  component='h4'
                  variant='caption1'
                  align='left'
                  spacing='0 0 0 12px'
                >
                  {element}
                </TypographyElement>
              </div>
              <Button
                variation={ButtonVariation.secondary}
                size={ButtonSize.big}
                height={31}
                width={115}
                label='Retirer'
                handleClick={() => deleteLearner(element)}
              />
            </div>
          )
        }
      </div>
      <hr className={line} />
      <div className={licenseButtons}>
        <Button
          variation={ButtonVariation.secondary}
          size={ButtonSize.big}
          width={153}
          height={54}
          handleClick={cancelEdit}
          label={Text.abortButton}
        />
        <Button
          disabled={!learners.length}
          variation={ButtonVariation.primary}
          size={ButtonSize.big}
          width={153}
          height={54}
          handleClick={() => {
            editEmails(email)
            if (email.length === 0) closeModal()
            cancelEdit()
          }}
          label={Text.confirmButton}
        />
      </div>
    </div>
  )
}
