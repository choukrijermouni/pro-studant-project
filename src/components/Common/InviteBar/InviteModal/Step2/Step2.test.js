import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import Step2 from '.'
configure({ adapter: new Adapter() })

describe('>>> Step2 --- Shallow Render Step2 component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<Step2 />)
    tree = renderer.create(
      <Step2 />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
