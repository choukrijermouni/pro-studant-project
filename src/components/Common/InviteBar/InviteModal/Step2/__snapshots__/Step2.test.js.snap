// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Step2 --- Shallow Render Step2 component +++ render correctly and match snapshot 1`] = `
<div>
  <div
    className="modal-0-2-1"
  >
    <p
      className="text-0-2-8 text-d0-0-2-9"
    >
      Modifier apprenants
    </p>
    <span
      className="search-0-2-4"
    >
      <div
        className="root-0-2-10 root-d0-0-2-14"
      >
        <div
          className="inputGroup-0-2-11"
        >
          <div
            className="inputGroup-0-2-16 inputGroup-d0-0-2-19"
          >
            <input
              className="input-0-2-17 input-d1-0-2-20"
              disabled={false}
              id="unique_ID_1"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
          <div
            className="icon-0-2-22 icon-d0-0-2-24"
          >
            <span
              className="icon-search-alph-iconfont iconStyle-0-2-23 iconStyle-d1-0-2-25"
            />
          </div>
        </div>
      </div>
    </span>
    <div
      className="emailsBox-0-2-7"
    >
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-26 avatarGroup-d0-0-2-27"
          >
            ys
          </span>
          <p
            className="text-0-2-28 text-d0-0-2-29"
          >
            choukri@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-30 button-d0-0-2-32"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-34 text-d0-0-2-35 text-0-2-31 text-d1-0-2-33"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-36 avatarGroup-d0-0-2-37"
          >
            md
          </span>
          <p
            className="text-0-2-38 text-d0-0-2-39"
          >
            Richard@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-40 button-d0-0-2-42"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-44 text-d0-0-2-45 text-0-2-41 text-d1-0-2-43"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-46 avatarGroup-d0-0-2-47"
          >
            ah
          </span>
          <p
            className="text-0-2-48 text-d0-0-2-49"
          >
            Richard@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-50 button-d0-0-2-52"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-54 text-d0-0-2-55 text-0-2-51 text-d1-0-2-53"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-56 avatarGroup-d0-0-2-57"
          >
            ag
          </span>
          <p
            className="text-0-2-58 text-d0-0-2-59"
          >
            ayman@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-60 button-d0-0-2-62"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-64 text-d0-0-2-65 text-0-2-61 text-d1-0-2-63"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-66 avatarGroup-d0-0-2-67"
          >
            ak
          </span>
          <p
            className="text-0-2-68 text-d0-0-2-69"
          >
            aymano@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-70 button-d0-0-2-72"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-74 text-d0-0-2-75 text-0-2-71 text-d1-0-2-73"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-76 avatarGroup-d0-0-2-77"
          >
            mam
          </span>
          <p
            className="text-0-2-78 text-d0-0-2-79"
          >
            marouan@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-80 button-d0-0-2-82"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-84 text-d0-0-2-85 text-0-2-81 text-d1-0-2-83"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-86 avatarGroup-d0-0-2-87"
          >
            cb
          </span>
          <p
            className="text-0-2-88 text-d0-0-2-89"
          >
            choukri@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-90 button-d0-0-2-92"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-94 text-d0-0-2-95 text-0-2-91 text-d1-0-2-93"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-96 avatarGroup-d0-0-2-97"
          >
            dm
          </span>
          <p
            className="text-0-2-98 text-d0-0-2-99"
          >
            djamal@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-100 button-d0-0-2-102"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-104 text-d0-0-2-105 text-0-2-101 text-d1-0-2-103"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-106 avatarGroup-d0-0-2-107"
          >
            hb
          </span>
          <p
            className="text-0-2-108 text-d0-0-2-109"
          >
            hamid@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-110 button-d0-0-2-112"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-114 text-d0-0-2-115 text-0-2-111 text-d1-0-2-113"
          >
            Retirer
          </p>
        </button>
      </div>
      <div
        className="emailRow-0-2-5"
      >
        <div
          className="alignAvatar-0-2-6"
        >
          <span
            className="avatarGroup-0-2-116 avatarGroup-d0-0-2-117"
          >
            sd
          </span>
          <p
            className="text-0-2-118 text-d0-0-2-119"
          >
            said@gmail.com
          </p>
        </div>
        <button
          className="button-0-2-120 button-d0-0-2-122"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-124 text-d0-0-2-125 text-0-2-121 text-d1-0-2-123"
          >
            Retirer
          </p>
        </button>
      </div>
    </div>
    <hr
      className="line-0-2-3"
    />
    <div
      className="licenseButtons-0-2-2"
    >
      <button
        className="button-0-2-126 button-d0-0-2-128"
        onClick={[Function]}
        type="button"
      >
         
        <p
          className="text-0-2-130 text-d0-0-2-131 text-0-2-127 text-d1-0-2-129"
        >
          Annuler
        </p>
      </button>
      <button
        className="button-0-2-132 button-d0-0-2-134"
        onClick={[Function]}
        type="button"
      >
         
        <p
          className="text-0-2-136 text-d0-0-2-137 text-0-2-133 text-d1-0-2-135"
        >
          Confirmer
        </p>
      </button>
    </div>
  </div>
</div>
`;
